Some changes to articles API and data model

caiosba opened the pull request:
## Description

Some changes to articles API and data model:

- It should be possible to create `ClaimDescription` without a `ProjectMedia`
- Adding a `team_id` field to `ClaimDescription`
- Expose `ClaimDescription.project_media_id` in GraphQL mutations
- Adding a new many-to-many relationship between explainers and items, through a new join model `ExplainItem`, and respective GraphQL type and mutations
- Expose `FactCheck.rating` in GraphQL mutations
- New GraphQL fields for `ProjectMedia`: `fact_check` and `explainers`
- New filters for `Team.explainers` GraphQL connection: `standalone` and `text`

References: CV2-4441, CV2-4626 and CV2-4627.

## How has this been tested?

Automated tests added for everything.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [x] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

melsawy commented on app/models/explainer_item.rb:
I prefer to use  ProjectMedia instead of Item to be consistent with other relations i.e `class ExplainerProjectMedia`

caiosba replied:
Yeah, in this case I ended up using `Item` to make more explicit that it's for the full cluster, not only the main `ProjectMedia`.

melsawy commented on db/migrate/20240619141452_allow_null_project_medias_to_claims.rb:
I suggest to merge this one into `20240619131714_add_team_id_to_claim_descriptions.rb`   migration file

caiosba replied:
Done!

melsawy reviewed (approved):
LGTM
I added a two comments and it's up to you to apply or skip these comment so I approved the PR.
although I think you should add a rake task to set `team_id` for existing ClaimDescription`

caiosba commented:
> LGTM I added a two comments and it's up to you to apply or skip these comment so I approved the PR. although I think you should add a rake task to set `team_id` for existing ClaimDescription`

Thanks Sawy, I addressed your comments - regarding the migration, I listed it in CV2-4441 so we can handle it separately.
